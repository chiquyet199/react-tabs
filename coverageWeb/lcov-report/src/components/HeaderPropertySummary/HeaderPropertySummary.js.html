<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/HeaderPropertySummary/HeaderPropertySummary.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/HeaderPropertySummary</a> HeaderPropertySummary.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.92% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.91% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/11</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import {connect} from "react-redux";
import {go} from "../../actions/navigation";
import {
  getPrimaryPropertyImage,
  getRentDueDate,
} from "../../utils/helpers/property";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {func, shape, string} from "prop-types";
import moment from "moment";
import {
  ArrowIcon,
  Background,
  FullDetailsLink,
  PropertyImage,
  PropertyAddress,
  PropertyRent,
  PropertySummary,
  RentalPayments,
  RentDueDate,
  VerticalStack,
} from "./HeaderPropertySummary.style";
&nbsp;
export const HeaderPropertySummary = ({goTo, lease, translations}) =&gt; {
  <span class="missing-if-branch" title="else path not taken" >E</span>if (lease) {
    const uri = getPrimaryPropertyImage(lease.property.propertyImages);
    const rentDueDate = moment(getRentDueDate(lease)).format("DD.MM.YYYY");
    const {
      property: {
        id,
        address: {line1, town, postCode},
      },
      rentAmount,
    } = lease;
&nbsp;
    return (
      &lt;Background&gt;
        &lt;PropertySummary&gt;
          &lt;PropertyImage source={{uri}} /&gt;
          &lt;VerticalStack&gt;
            &lt;RentalPayments&gt;
              &lt;PropertyRent&gt;
                {translations.currency_symbol}
                {rentAmount}
              &lt;/PropertyRent&gt;
              &lt;RentDueDate&gt;
                {`${translations.rentDueDate} ${rentDueDate}`}
              &lt;/RentDueDate&gt;
            &lt;/RentalPayments&gt;
            &lt;PropertyAddress&gt;
              {`${line1}, ${town}, ${postCode}`}
            &lt;/PropertyAddress&gt;
          &lt;/VerticalStack&gt;
          &lt;FullDetailsLink onPress={() =&gt; goTo(`/property-listing/${id}`)}&gt;
            &lt;ArrowIcon name="arrow-right" size={32} /&gt;
          &lt;/FullDetailsLink&gt;
        &lt;/PropertySummary&gt;
      &lt;/Background&gt;
    );
  }
&nbsp;
<span class="cstat-no" title="statement not covered" >  return null;</span>
};
&nbsp;
HeaderPropertySummary.propTypes = {
  goTo: func.isRequired,
  lease: LeaseSpec,
  translations: shape({
    currency_symbol: string.isRequired,
    rentDueDate: string.isRequired,
  }),
};
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  translations: {
    currency_symbol: state.locale.translations.common.currency_symbol,
    ...state.locale.translations.common.headerPropertySummary,
  },
});
&nbsp;
const mapDisatchToProps = <span class="fstat-no" title="function not covered" >()</span> =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  goTo: go,
});
&nbsp;
export default connect(
  mapStateToProps,
  mapDisatchToProps,
)(HeaderPropertySummary);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
