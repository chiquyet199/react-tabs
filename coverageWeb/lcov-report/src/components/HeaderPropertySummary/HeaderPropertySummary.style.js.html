<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/HeaderPropertySummary/HeaderPropertySummary.style.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/HeaderPropertySummary</a> HeaderPropertySummary.style.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/16</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {themed} from "../../utils/common";
import Icon from "../Icon";
import styledNative from "styled-components/native";
&nbsp;
export const Background = themed(styledNative.View`
  background-color: ${<span class="fstat-no" title="function not covered" >({</span>globaltheme, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    theme[globaltheme].topbar.background}</span>;
`);
export const PropertySummary = themed(styledNative.View`
  align-items: center;
  background-color: ${<span class="fstat-no" title="function not covered" >({</span>globaltheme, theme}) =&gt;
<span class="cstat-no" title="statement not covered" >    theme[globaltheme].activeTenancy.propertySummaryBackground}</span>;
  border-radius: 6px;
  flex-direction: row;
  padding: 10px;
  margin: 12px 33px 11px 18px;
`);
export const PropertyImage = styledNative.Image`
  border-radius: 6px;
  height: 56;
  margin-right: 12px;
  width: 56;
`;
export const ThemedText = themed(styledNative.Text`
  color: ${<span class="fstat-no" title="function not covered" >({</span>globaltheme, theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme[globaltheme].altTextColor}</span>;
`);
export const PropertyRent = styledNative(ThemedText)`
  font-size: 24;
  margin-right: 5;
`;
export const RentDueDate = styledNative(ThemedText)`
  font-size: 12;
`;
export const PropertyAddress = styledNative(ThemedText)`
  font-size: 12;
  font-weight: bold;
`;
export const FullDetailsLink = themed(styledNative.TouchableOpacity`
  border-radius: 16px;
  backgroundColor: ${<span class="fstat-no" title="function not covered" >({</span>globaltheme, theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme[globaltheme].accentColor}</span>;
  position: relative;
  left: 13;
`);
export const ArrowIcon = themed(styledNative(Icon)`
  color: ${<span class="fstat-no" title="function not covered" >({</span>globaltheme, theme}) =&gt; <span class="cstat-no" title="statement not covered" >theme[globaltheme].altTextColor}</span>;
`);
export const VerticalStack = styledNative.View`
&nbsp;
`;
export const RentalPayments = styledNative.View`
  align-items: flex-end;
  flex-direction: row;
  margin-bottom: 4px;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
