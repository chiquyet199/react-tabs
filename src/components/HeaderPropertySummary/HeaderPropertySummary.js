import React from "react";
import {connect} from "react-redux";
import {go} from "../../actions/navigation";
import {
  getPrimaryPropertyImage,
  getRentDueDate,
} from "../../utils/helpers/property";
import LeaseSpec from "../../utils/api/specs/LeaseSpec";
import {func, shape, string} from "prop-types";
import moment from "moment";
import {
  ArrowIcon,
  Background,
  FullDetailsLink,
  PropertyImage,
  PropertyAddress,
  PropertyRent,
  PropertySummary,
  RentalPayments,
  RentDueDate,
  VerticalStack,
} from "./HeaderPropertySummary.style";

export const HeaderPropertySummary = ({goTo, lease, translations}) => {
  if (lease) {
    const uri = getPrimaryPropertyImage(lease.property.propertyImages);
    const rentDueDate = moment(getRentDueDate(lease)).format("DD.MM.YYYY");
    const {
      property: {
        id,
        address: {line1, town, postCode},
      },
      rentAmount,
    } = lease;

    return (
      <Background>
        <PropertySummary>
          <PropertyImage source={{uri}} />
          <VerticalStack>
            <RentalPayments>
              <PropertyRent>
                {translations.currency_symbol}
                {rentAmount}
              </PropertyRent>
              <RentDueDate>
                {`${translations.rentDueDate} ${rentDueDate}`}
              </RentDueDate>
            </RentalPayments>
            <PropertyAddress>
              {`${line1}, ${town}, ${postCode}`}
            </PropertyAddress>
          </VerticalStack>
          <FullDetailsLink onPress={() => goTo(`/property-listing/${id}`)}>
            <ArrowIcon name="arrow-right" size={32} />
          </FullDetailsLink>
        </PropertySummary>
      </Background>
    );
  }

  return null;
};

HeaderPropertySummary.propTypes = {
  goTo: func.isRequired,
  lease: LeaseSpec,
  translations: shape({
    currency_symbol: string.isRequired,
    rentDueDate: string.isRequired,
  }),
};

const mapStateToProps = state => ({
  translations: {
    currency_symbol: state.locale.translations.common.currency_symbol,
    ...state.locale.translations.common.headerPropertySummary,
  },
});

const mapDisatchToProps = () => ({
  goTo: go,
});

export default connect(
  mapStateToProps,
  mapDisatchToProps,
)(HeaderPropertySummary);
