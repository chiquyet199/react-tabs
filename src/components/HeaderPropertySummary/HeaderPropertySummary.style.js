import {themed} from "../../utils/common";
import Icon from "../Icon";
import styledNative from "styled-components/native";

export const Background = themed(styledNative.View`
  background-color: ${({globaltheme, theme}) =>
    theme[globaltheme].topbar.background};
`);
export const PropertySummary = themed(styledNative.View`
  align-items: center;
  background-color: ${({globaltheme, theme}) =>
    theme[globaltheme].activeTenancy.propertySummaryBackground};
  border-radius: 6px;
  flex-direction: row;
  padding: 10px;
  margin: 12px 33px 11px 18px;
`);
export const PropertyImage = styledNative.Image`
  border-radius: 6px;
  height: 56;
  margin-right: 12px;
  width: 56;
`;
export const ThemedText = themed(styledNative.Text`
  color: ${({globaltheme, theme}) => theme[globaltheme].altTextColor};
`);
export const PropertyRent = styledNative(ThemedText)`
  font-size: 24;
  margin-right: 5;
`;
export const RentDueDate = styledNative(ThemedText)`
  font-size: 12;
`;
export const PropertyAddress = styledNative(ThemedText)`
  font-size: 12;
  font-weight: bold;
`;
export const FullDetailsLink = themed(styledNative.TouchableOpacity`
  border-radius: 16px;
  backgroundColor: ${({globaltheme, theme}) => theme[globaltheme].accentColor};
  position: relative;
  left: 13;
`);
export const ArrowIcon = themed(styledNative(Icon)`
  color: ${({globaltheme, theme}) => theme[globaltheme].altTextColor};
`);
export const VerticalStack = styledNative.View`

`;
export const RentalPayments = styledNative.View`
  align-items: flex-end;
  flex-direction: row;
  margin-bottom: 4px;
`;
