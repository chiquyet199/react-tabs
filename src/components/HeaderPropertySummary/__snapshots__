/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/HeaderPropertySummary/__native_snapshots__