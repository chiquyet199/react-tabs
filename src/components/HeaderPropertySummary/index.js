import HeaderPropertySummary from "./HeaderPropertySummary";

export default HeaderPropertySummary;
