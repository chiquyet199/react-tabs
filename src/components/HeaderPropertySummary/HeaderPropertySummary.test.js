import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import {HeaderPropertySummary} from "./HeaderPropertySummary";
import {FullDetailsLink} from "./HeaderPropertySummary.style";
import {
  getRentDueDate,
  getPrimaryPropertyImage,
} from "../../utils/helpers/property";

jest.mock("../../utils/helpers/property");

describe("HeaderPropertySummary test", () => {
  beforeAll(() => {
    getRentDueDate.mockReturnValue("2019-01-07");
    getPrimaryPropertyImage.mockReturnValue(
      "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
    );
  });

  beforeEach(() => {
    getRentDueDate.mockClear();
    getPrimaryPropertyImage.mockClear();
  });

  it(`HeaderPropertySummary snapshot test on - ${Platform.OS}`, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("should pass lease to getRentDueDate", () => {
    const lease = getMockLease();
    getComponent({lease});
    expect(getRentDueDate).toHaveBeenCalledWith(lease);
  });

  it("should pass property images to getPrimaryPropertyImage", () => {
    const property = getMockProperty();
    getComponent({lease: getMockLease(property)});
    expect(getPrimaryPropertyImage).toHaveBeenCalledWith(
      property.propertyImages,
    );
  });

  it("calls goTo with '/property-listing/{id}' when FullDetailsLink pressed", () => {
    const propertyId = "abc-123-def-456";
    const lease = getMockLease();
    lease.property.id = propertyId;
    const goTo = jest.fn();
    const component = getComponent({goTo, lease});
    component.find(FullDetailsLink).simulate("press");
    expect(goTo).toHaveBeenCalledWith(`/property-listing/${propertyId}`);
  });

  const getComponent = props => {
    const parsedProps = {
      goTo: () => {},
      lease: getMockLease(),
      translations: {
        currency_symbol: "currency_symbol",
        rentDueDate: "rentDueDate",
      },
      ...props,
    };

    return shallow(<HeaderPropertySummary {...parsedProps} />);
  };
});

const getMockLease = (property = getMockProperty()) => ({
  id: "abc-123",
  branchId: "def-456",
  property,
  renters: [
    {
      userId: "ghi-789",
      depositOptions: [{type: "DEPOSIT_FREE"}],
    },
  ],
  status: "ACTIVE",
  rentAmount: 1234,
  rentFrequency: "MONTHLY",
  duration: {
    minimumMonths: 6,
    startDate: "2018-11-06T00:00:00",
  },
  conditionsFreeText: "",
  documents: [],
});

const getMockProperty = () => ({
  id: "abc-123",
  presentation: {
    bathroomsCount: 1,
    bedroomsCount: 2,
    furnishedType: "PART_FURNISHED",
    hasOutsideSpace: true,
    hasParking: true,
    petsAllowed: false,
    propertyType: "HOUSE",
    smokingAllowed: false,
  },
  rentFrequency: "WEEKLY",
  rentAmount: 1234.99,
  depositAmount: 1000,
  depositFreeAvailable: true,
  propertyImages: [
    {
      id: "def-456",
      url: "http://www.findyourcanopy.com/img123.jpeg",
      isPrimary: false,
    },
    {
      id: "def-457",
      url:
        "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
      isPrimary: true,
    },
    {
      id: "def-458",
      url: "http://www.findyourcanopy.com/img125.jpeg",
      isPrimary: false,
    },
  ],
  landlord: {},
  address: {
    line1: "29 Acacia Road",
    town: "Beanotown",
    state: "Aberdeenshire",
    postCode: "BE1 1BE",
    countryCode: "UK",
  },
  createdBy: "Homogenous cohorts",
  agencyId: "22",
  branchId: "44",
});
