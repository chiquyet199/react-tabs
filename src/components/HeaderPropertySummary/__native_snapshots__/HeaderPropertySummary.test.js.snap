// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderPropertySummary test HeaderPropertySummary snapshot test on - ios 1`] = `
<Connect(WithTheme(Styled(Component)))>
  <Connect(WithTheme(Styled(Component)))>
    <Styled(Image)
      source={
        Object {
          "uri": "https://media.rightmove.co.uk/dir/152k/151109/61705537/151109_344038-1_IMG_19_0002_max_656x437.jpg",
        }
      }
    />
    <Styled(Component)>
      <Styled(Component)>
        <Styled(Connect(WithTheme(Styled(Text))))>
          currency_symbol
          1234
        </Styled(Connect(WithTheme(Styled(Text))))>
        <Styled(Connect(WithTheme(Styled(Text))))>
          rentDueDate 07.01.2019
        </Styled(Connect(WithTheme(Styled(Text))))>
      </Styled(Component)>
      <Styled(Connect(WithTheme(Styled(Text))))>
        29 Acacia Road, Beanotown, BE1 1BE
      </Styled(Connect(WithTheme(Styled(Text))))>
    </Styled(Component)>
    <Connect(WithTheme(Styled(TouchableOpacity)))
      onPress={[Function]}
    >
      <Connect(WithTheme(Styled(Icon)))
        name="arrow-right"
        size={32}
      />
    </Connect(WithTheme(Styled(TouchableOpacity)))>
  </Connect(WithTheme(Styled(Component)))>
</Connect(WithTheme(Styled(Component)))>
`;
